server.go

```diff
@@ -1823,6 +1833,25 @@ func (s *server) UpdateRoutingConfig(cfg *routing.MissionControlConfig) {
 	routerCfg.MaxMcHistory = cfg.MaxMcHistory
 }
 
+// registerBlockConsumers registers the subsystems that consume block events.
+// By calling `RegisterQueue`, a list of subsystems are registered in the
+// blockbeat for block notifications. When a new block arrives, the subsystems
+// in the same queue are notified sequentially, and different queues are
+// notified concurrently.
+//
+// NOTE: To put a subsystem in a different queue, create a slice and pass it to
+// a new `RegisterQueue` call.
+func (s *server) registerBlockConsumers() {
```

Comment:
I find it more intuitive to add the consumers as a func paramter here, and have the consumer record defined before calling this function.

Reply:
can you elaborate?

Reply:
something like this but just my perspective.

`func (s *server) registerBlockConsumers( []chainio.Consumer) {`